### PR TITLE
Enhance support for nil values

### PR DESCRIPTION
**Pull request by kilnerm**
This PR adds support for using nil to represent NULL for default column values.

The PR also adds test for inset, update and table creation when using nil values.

### CODE REVIEW

File: Sources/SwiftKuery/ColumnCreator.swift [hunk 1]
```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
     /// - Parameter column: The column being built
     /// - Returns: A string representation of the column for the implementing database or nil if it cannot be built
     func buildColumn(for column: Column, using queryBuilder: QueryBuilder) -> String?
+
+    /// Build an appropriate representation of a passed value for the database
+    /// A default implemetation is provided that can be overriden should a
```
Comment (pushkarnk) [inline Sources/SwiftKuery/ColumnCreator.swift]: typo: `implemetation`

File: Sources/SwiftKuery/ColumnCreator.swift [hunk 2]
```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
...
+        case let val as Parameter:
+            return try val.build(queryBuilder: queryBuilder)
+        case let value as Date:
+            if let dateFormatter = queryBuilder.dateFormatter {
+                return dateFormatter.string(from: value)
+            }
+            return "'\(String(describing: value))'"
+        default:
+            let val = String(describing: item)
+            return val == "nil" ? "NULL" : val
```
Comment (pushkarnk) [inline Sources/SwiftKuery/ColumnCreator.swift]: Do we need the `if column.nullDefaultValue` check here?

File: Sources/SwiftKuery/ColumnCreator.swift [hunk 3]
```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
...
+
+    /// Build an appropriate representation of a passed value for the database
+    /// A default implemetation is provided that can be overriden should a
+    /// plugin require non-common behaviour
+    ///
+    /// - Parameter item: The value to convert
+    /// - Returns: A string representing the value that can be passed into the database
+    func packType(_ item: Any, queryBuilder: QueryBuilder) throws -> String
+
+    /// Get the dafault value for a column
```
Comment (pushkarnk) [inline Sources/SwiftKuery/ColumnCreator.swift]: typo: `dafault`

File: Sources/SwiftKuery/ColumnCreator.swift (hunk 2 above)
Comment (pushkarnk) [reply Sources/SwiftKuery/ColumnCreator.swift]: In the current scheme of things, we'll reach here only if a `defaultValue` is present but we aren't able to represent it as a String. I'm not considering scenarios where `packType` will be called from sites outside of `getDefaultValue`.

Comment (kilnerm) [reply Sources/SwiftKuery/ColumnCreator.swift]: Given we check the passed parameter is not nil prior to calling this function, see `getDefaultValue`, I don't believe we need to return anything other than `String(describing: item)` here.
